packages/contract/lib/reason.js

```diff
@@ -23,14 +32,28 @@ const reason = {
       res && typeof res === "object" && typeof res.result === "string";
 
     if (isObject) {
+      // TODO: ganache 2.0 returns the reason string already when
+      // vmErrorsOnRPCResponse===true. Why not use it?
       const data = res.error.data;
-      const hash = Object.keys(data)[0];
+      let resData;
+      if ("result" in data) {
+        // there is a single result (only in ganache 3.0+)
+        resData = data.result;
+      } else {
+        // handle `evm_mine`, `miner_start`, batch payloads, and ganache 2.0
+        // TODO: there are 1 or more failed transactions, handle multiple?
+        const hash = Object.keys(data)[0];
```

Comment:
In ganache v2, and ganache v3 when used with miner_start, evm_mine, and batched payloads, the error's `data` property is a mapping of transaction hashes to error details.